src/osd/PrimaryLogPG.cc

```diff
@@ -2344,6 +2344,17 @@ PrimaryLogPG::cache_result_t PrimaryLogPG::maybe_handle_manifest_detail(
     }
     return cache_result_t::HANDLED_PROXY;
   case object_manifest_t::TYPE_CHUNKED:
+    if (obc->obs.oi.size == 0) {
+      const MOSDOp *m = static_cast<const MOSDOp*>(op->get_req());
+      const object_locator_t oloc = m->get_object_locator();
+      promote_object(obc, obc->obs.oi.soid, oloc, op, NULL);
+      return cache_result_t::BLOCKED_PROMOTE;
+    }
+    if (can_proxy_chunked_read(op)) {
+      do_proxy_chunked_op(op, obc->obs.oi.soid, obc, write_ordered);
+      return cache_result_t::HANDLED_PROXY;
+    }
```

Comment:
can_proxy_chunked_read() just checks whether OPs (such as CEPH_OSD_OP_READ..)
can be handled or not. If you look at the code above, you can see if (oi.size == 0) {  promote_object() } 
oi.size == 0 indicate that the object need to be promoted (== can't proxy).